### DISCUSSION TITLE
Do you have any plans to support Java data processing?

### DISCUSSION THREAD

**Discussion by detongz** [apache/seatunnel#1560] [Q&A] [answered]
In docs says seatunnel supports SQL for data processing. Are you planning to support like load java jars for data processing?

**Comment by ruanwenjun** [comment, marked as answer]
Cool, right now, seatunnel support Flink and Spark as the runtime engine, this means seatunnel will be packaged as a Flink or Spark job running in Flink or Spark cluster.

In my knowledge, it might be hard to load the jar written by users, maybe you need to put the jar in Flink/Spark lib directory?

**Reply by seatunneltrailuser** [reply]
Jobs submitted thru APIs using version v2.3.3, does'nt get listed in Flink UI, but whenever we use the Shell script starter (Flink 15), it is getting listed in Flink UI and get executed successfully also. 

Are we missing any additional configuration or environment variables or job parameters or even explicit setting in env attribute of Request body ?
